File: financial-domain-starter/src/main/java/com/nixsolutions/financial/discovery/ServiceDiscoveryProperties.java
package com.nixsolutions.financial.discovery;

import java.util.ArrayList;
import java.util.List;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.validation.annotation.Validated;
import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
@ConfigurationProperties(prefix = "services")
@Validated
public class ServiceDiscoveryProperties
{
  private List<ServiceRecord> list = new ArrayList<>();
}

File: financial-domain-starter/src/main/kotlin/com/nixsolutions/financial/discovery/ServiceDiscoveryProperties.kt
package com.nixsolutions.financial.discovery

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated
import java.util.*

@ConfigurationProperties(prefix = "services")
@Validated
class ServiceDiscoveryProperties {
  var list: List<ServiceRecord> = ArrayList();
}
